include/Models/Language_Model.php

```diff
@@ -0,0 +1,1107 @@
+					}
+				}
+			}
+		}
+
+		// Delete relationships
+		if ( ! empty( $dr ) ) {
+			// PHPCS:disable WordPress.DB.PreparedSQL.NotPrepared
+			$wpdb->query(
+				"DELETE FROM $wpdb->term_relationships
+				WHERE object_id IN ( " . implode( ',', $dr['id'] ) . ' )
+				AND term_taxonomy_id IN ( ' . implode( ',', $dr['tt'] ) . ' )'
+			);
+			// PHPCS:enable
+		}
+
+		// Delete terms
+		if ( ! empty( $dt ) ) {
+			$wpdb->query( "DELETE FROM $wpdb->terms WHERE term_id IN ( " . implode( ',', $dt['t'] ) . ' )' ); // PHPCS:ignore WordPress.DB.PreparedSQL.NotPrepared
+			$wpdb->query( "DELETE FROM $wpdb->term_taxonomy WHERE term_taxonomy_id IN ( " . implode( ',', $dt['tt'] ) . ' )' ); // PHPCS:ignore WordPress.DB.PreparedSQL.NotPrepared
+		}
+
+		// Update terms
+		if ( ! empty( $ut ) ) {
+			// PHPCS:disable WordPress.DB.PreparedSQL.NotPrepared
+			$wpdb->query(
+				"UPDATE $wpdb->term_taxonomy
+				SET description = ( CASE term_id " . implode( ' ', $ut['case'] ) . ' END )
+				WHERE term_id IN ( ' . implode( ',', $ut['in'] ) . ' )'
+			);
+			// PHPCS:enable
+		}
+
+		if ( ! empty( $term_ids ) ) {
+			foreach ( $term_ids as $taxonomy => $ids ) {
+				clean_term_cache( $ids, $taxonomy );
+			}
+		}
+	}
+
+	/**
+	 * Maybe adds the missing language terms for 3rd party language taxonomies.
+	 *
+	 * @since 3.4
+	 * @since 3.7 Moved from `PLL_Model::maybe_create_language_terms()` to `WP_Syntex\Polylang\Models\Language_Model::maybe_create_language_terms()`.
+	 *
+	 * @return void
+	 */
+	public function maybe_create_language_terms(): void {
+		$registered_taxonomies = array_diff(
+			$this->translatable_objects->get_taxonomy_names( array( 'language' ) ),
+			// Exclude the post and term language taxonomies from the list.
+			array(
+				$this->translatable_objects->get( 'post' )->get_tax_language(),
+				$this->translatable_objects->get( 'term' )->get_tax_language(),
+			)
+		);
+
+		if ( empty( $registered_taxonomies ) ) {
+			// No 3rd party language taxonomies.
+			return;
+		}
+
+		// We have at least one 3rd party language taxonomy.
+		$known_taxonomies = $this->options['language_taxonomies'];
+		$new_taxonomies   = array_diff( $registered_taxonomies, $known_taxonomies );
+
+		if ( empty( $new_taxonomies ) ) {
+			// No new 3rd party language taxonomies.
+			return;
+		}
+
+		// We have at least one unknown 3rd party language taxonomy.
+		foreach ( $this->get_languages_list() as $language ) {
+			$this->update_secondary_language_terms( $language->slug, $language->name, $language, $new_taxonomies );
+		}
+
+		// Clear the cache, so the new `term_id` and `term_taxonomy_id` appear in the languages list.
+		$this->clean_cache();
+
+		// Keep the previous values, so this is triggered only once per taxonomy.
+		$this->options['language_taxonomies'] = array_merge( $known_taxonomies, $new_taxonomies );
+	}
+
+	/**
+	 * Cleans language cache.
+	 *
+	 * @since 3.7
+	 * @return void
+	 */
+	public function clean_cache(): void {
```

Comment:
This method should have the same signature as `PLL_Model::clean_languages_cahce()` IMHO.
So we'll be able to deprecate the other and use this one instead. What do you think?

Reply:
Personally I prefer without parameters, so it doesn't depend on a WP hook: this way, the class doesn't mix hooks and processes.

Reply:
I get your point 👍